prov/rxm/src/rxm_ep.c

```diff
@@ -1152,14 +1159,25 @@ rxm_ep_inject_common(struct rxm_ep *rxm_ep, const void *buf, size_t len,
 	rxm_conn = rxm_acquire_conn(rxm_ep, dest_addr);
 	if (OFI_UNLIKELY(rxm_conn->handle.state != CMAP_CONNECTED)) {
 		ret = rxm_ep_handle_unconnected(rxm_ep, &rxm_conn->handle, dest_addr);
-		if (!ret)
+		if (!ret) {
 			goto inject_continue;
+		} else if (OFI_UNLIKELY(ret != -FI_EAGAIN)) {
+			fastlock_release(&rxm_ep->util_ep.cmap->lock);
+			return ret;
+		}
+
 		fastlock_release(&rxm_ep->util_ep.cmap->lock);
-		return ret;
+		goto defer;
 	}
 inject_continue:
 	fastlock_release(&rxm_ep->util_ep.cmap->lock);
 
+	if (OFI_UNLIKELY(!dlist_empty(&rxm_conn->deferred_op_list))) {
+		rxm_ep_progress_multi(&rxm_ep->util_ep);
+		if (!dlist_empty(&rxm_conn->deferred_op_list))
+			goto defer;
+	}
+
```

Comment:
Can the above be subsumed into `rxm_acquire_conn`?

Reply:
I would try to keep rxm_acquire_conn() limited in what it does.  Processing a deferred work list seems like a side effect.

Reply:
Yes, I was hoping rxm_acquire_conn only subsume those parts that are relevant, like handling unconnected endpoint.